### PR TITLE
Update kotlinx.coroutines.test to 1.6

### PR DESCRIPTION
**Pull request by zsmb13**

### CODE REVIEW

**Comment by zsmb13** [comment]
API 29 is failing inexplicably on CI. All green locally on both 29 and 32, it's just some kind of emulator problem on CI.

File: mobile/src/test/java/com/google/samples/apps/iosched/ui/schedule/ScheduleViewModelTest.kt [hunk 1]
```diff
@@ -382,7 +382,7 @@ class ScheduleViewModelTest {
             viewModel.scrollToStartOfDay(TestData.TestConferenceDays[0])
 
             val result = mutableListOf<ScheduleScrollEvent>()
-            val job = launch {
+            val job = launch(UnconfinedTestDispatcher()) {
```
Comment (manuelvicnt) [inline mobile/src/test/java/com/google/samples/apps/iosched/ui/schedule/ScheduleViewModelTest.kt:385]: Nit: Why not using the dispatcher from the coroutine rule?

Comment (zsmb13) [reply mobile/src/test/java/com/google/samples/apps/iosched/ui/schedule/ScheduleViewModelTest.kt:385]: It doesn't really matter whether you reuse the rule's dispatcher or create a new one, if both are of the same type (`UnconfinedTestDispatcher`, in this case). This way it's more obvious why a dispatcher is being passed in to `launch` - to have the coroutine run eagerly.

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #Wed Feb 10 08:38:31 CET 2021
 distributionBase=GRADLE_USER_HOME
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.0.2-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.2-bin.zip
```
Comment (JoseAlcerreca) [inline gradle/wrapper/gradle-wrapper.properties:3]: you need to update gradlew too

Comment (zsmb13) [reply gradle/wrapper/gradle-wrapper.properties]: Ah good catch, thank you, I actually forgot that. Must've updated this version automatically using Studio 🤔

**Review by JoseAlcerreca — APPROVED** [review]

**Review by manuelvicnt — APPROVED** [review]

**Pull request merged**